### PR TITLE
Format packages.json

### LINKED ISSUE #84: Need a command to reformat package set file automatically
I have something like this in my add-from-bower PR, but we really explicitly need one for when one works with package sets to make sure everyone actually edits these correctly in alphabetical order without inconsistent formatting, since JSON is... not fun.

### PR DESCRIPTION
**Pull request by justinwoo**
Based off of add-from-bower because of laziness and the assumption that we will merge it soon. Look at only the commit if desired.

This PR does really nothing much, but this would really help a lot when working with packages.json and with our diffs.

Example of this in action: https://i.imgur.com/d9L6Nv8.png

Fixes #84

### CODE REVIEW

**Comment by justinwoo** [comment]
Why I really want this exhibit A: https://github.com/purescript/package-sets/pull/143/files

**Comment by justinwoo** [comment]
This PR needs someone to review and approve :smile:

File: app/Main.hs [hunk 1]
```diff
@@ -518,6 +518,15 @@ addFromBower name = do
...
     mkPackageName' = Bifunctor.first show . mkPackageName . stripBowerNamePrefix
 
+formatPackageFile :: IO ()
+formatPackageFile = do
+    pkg <- readPackageFile
+    db <- readPackageSet pkg
+    writePackageSet' db
+  where
+    writePackageSet' =
+      writeTextFile "packages.json" . packageSetToJSON
```
Comment (kritzcreek) [inline app/Main.hs]: If we have a `readPackageFile` should we maybe also have a `writePackageFile`? Then you could define "packages.json" as a constant and use it in both definitions.

File: app/Main.hs [hunk 2]
```diff
@@ -518,6 +518,15 @@ addFromBower name = do
     stripBowerNamePrefix s = fromMaybe s $ T.stripPrefix "purescript-" s
     mkPackageName' = Bifunctor.first show . mkPackageName . stripBowerNamePrefix
 
+formatPackageFile :: IO ()
+formatPackageFile = do
+    pkg <- readPackageFile
```
Comment (kritzcreek) [inline app/Main.hs]: Totally up to you, but I don't think `pkg` and `db` are great names here, how about doing it pointfree?

`readPackageFile >>= readPackageSet >>= writePackageSet`

File: app/Main.hs (hunk 1 above)
Comment (justinwoo) [reply app/Main.hs]: we have a `writePackageFile`, but it writes to the psc-package.json, and then we have `readPackageSet` but it writes to packages.json within the package set you're using currently

File: app/Main.hs (hunk 2 above)
Comment (justinwoo) [reply app/Main.hs]: Sure, just mirrors some other parts but that's fine for me

File: app/Main.hs (hunk 1 above)
Comment (justinwoo) [reply app/Main.hs]: I guess in reality I should really be reading the packages.json file too though, yeah

**Review by kritzcreek — APPROVED** [review]

**Pull request merged**